scripts/src/javatests/org/oppia/android/scripts/ci/RetrieveChangedFilesTest.kt

```diff
@@ -0,0 +1,290 @@
+    originalStandardOutputStream = System.out
+    System.setOut(PrintStream(pendingOutputStream))
+  }
+
+  @After
+  fun tearDown() {
+    // Reinstate test output redirection.
+    System.setOut(PrintStream(pendingOutputStream))
+
+    // Print the status of the git repository to help with debugging in the cases of test failures
+    // and to help manually verify the expect git state at the end of each test.
+    println("git status (at end of test):")
+
+    scriptBgDispatcher.close()
+  }
+
+  @Test
+  fun testUtility_noArguments_printsUsageStringAndExits() {
+    val exception = assertThrows<SecurityException>() { runScript() }
+
+    // Bazel catches the System.exit() call and throws a SecurityException. This is a bit hacky way
+    // to verify that System.exit() is called, but it's helpful.
+    assertThat(exception).hasMessageThat().contains("System.exit()")
+    assertThat(pendingOutputStream.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testUtility_oneArgument_printsUsageStringAndExits() {
+    val exception = assertThrows<SecurityException>() { runScript("arg1") }
+
+    // Bazel catches the System.exit() call and throws a SecurityException. This is a bit hacky way
+    // to verify that System.exit() is called, but it's helpful.
+    assertThat(exception).hasMessageThat().contains("System.exit()")
+    assertThat(pendingOutputStream.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testUtility_twoArguments_printsUsageStringAndExits() {
+    val exception = assertThrows<SecurityException>() { runScript("arg1", "arg2") }
+
+    // Bazel catches the System.exit() call and throws a SecurityException. This is a bit hacky way
+    // to verify that System.exit() is called, but it's helpful.
+    assertThat(exception).hasMessageThat().contains("System.exit()")
+    assertThat(pendingOutputStream.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testUtility_threeArguments_printsUsageStringAndExits() {
+    val exception = assertThrows<SecurityException>() { runScript("arg1", "arg2", "arg3") }
+
+    // Bazel catches the System.exit() call and throws a SecurityException. This is a bit hacky way
+    // to verify that System.exit() is called, but it's helpful.
+    assertThat(exception).hasMessageThat().contains("System.exit()")
+    assertThat(pendingOutputStream.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testUtility_fourArguments_printsUsageStringAndExits() {
+    val exception = assertThrows<SecurityException>() { runScript("arg1", "arg2", "arg3", "arg4") }
+
+    // Bazel catches the System.exit() call and throws a SecurityException. This is a bit hacky way
+    // to verify that System.exit() is called, but it's helpful.
+    assertThat(exception).hasMessageThat().contains("System.exit()")
+    assertThat(pendingOutputStream.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testUtility_invalidBase64_throwsException() {
+    assertThrows<IllegalArgumentException>() {
```

Comment:
Can the cause message of the exception be verified?

Reply:
Sure, added the following:

```kotlin
// Indicates the Proto data is improperly formatted or corrupted.
assertThat(exception).hasMessageThat().contains(
  "Last unit does not have enough valid bits"
)
```